packages/xod-client-electron/src/view/containers/App.jsx

```diff
@@ -224,6 +226,7 @@ class App extends client.App {
           R.reject(R.isNil)
         )([payload.message, payload.error.stdout, payload.stderr]);
         proc.fail(failureMessage, payload.percentage);
+        this.props.actions.updateCompileLimit();
       }
       // Remove listener if process is finished.
       ipcRenderer.removeAllListeners(UPLOAD_TO_ARDUINO);
```

Comment:
Looks like you don't need to call `updateCompileLimit` twice in both conditions, just call it once here.